### PR TITLE
Add a simulator for cumulative reward

### PR DESCRIPTION
**Pull request by altineel**
The current structure of the simulation returns the reward for the last action node. However, the maritime logistic implementation requires a cumulative reward function. I believe this update wouldn't cause a problem if the intermittent state transitions have reward = 0. This update, for example, can be applicable to GridWorld example.

### CODE REVIEW

File: mcts4py/GenericSolver.py [hunk 1]
```diff
@@ -101,6 +101,44 @@ def simulate(self, node: ActionNode[TState, TAction]) -> float:
...
+            reward = self.mdp.reward(current_state, random_action, new_state) * discount
+            if self.verbose:
+                print(f"-> Depth limit reached: {reward}")
+            return reward
+        next_node = ActionNode(node, random_action)
+        next_node.state = new_state
+        reward += self.simulate(next_node, depth=depth + 1)
+        return reward
+
+
```
Comment (larkz) [inline mcts4py/GenericSolver.py:141]: Can you add a test case for this method in a separate test file?

**Review by JunTaoLuo — APPROVED** [review]

File: mcts4py/GenericSolver.py (hunk 1 above)
Comment (altineel) [reply mcts4py/GenericSolver.py:141]: I'm not sure what kind of structure I should be building for the test case. Could you maybe give me an example ?

Comment (larkz) [reply mcts4py/GenericSolver.py:141]: You can do this separately in another PR, perhaps I will discuss with you in person on this, for now I will merge the changes, and release an update to mcts4py.

**Comment by larkz** [comment]
Changes have been released and can be installed using,
```
python3 -m pip install --index-url https://test.pypi.org/simple/ mcts4py-0.16
```

**Comment by altineel** [comment]
> Changes have been released and can be installed using,
> 
> ```
> python3 -m pip install --index-url https://test.pypi.org/simple/ mcts4py-0.16
> ```
> 
> @altineel

Unfortunately I get the below error. In fact I got this error before this update, so I included the project into the other one but this needs to be fixed any way. Do you have any idea ?  @larkz 

```
pip install --index-url https://test.pypi.org/simple/ mcts4py-0.16
Looking in indexes: https://test.pypi.org/simple/
ERROR: Could not find a version that satisfies the requirement mcts4py-0.16 (from versions: none)
ERROR: No matching distribution found for mcts4py-0.16
```

**Comment by larkz** [comment]
> > Changes have been released and can be installed using,
> > ```
> > python3 -m pip install --index-url https://test.pypi.org/simple/ mcts4py-0.16
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > @altineel
> 
> Unfortunately I get the below error. In fact I got this error before this update, so I included the project into the other one but this needs to be fixed any way. Do you have any idea ?  @larkz
> 
> ```
> pip install --index-url https://test.pypi.org/simple/ mcts4py-0.16
> Looking in indexes: https://test.pypi.org/simple/
> ERROR: Could not find a version that satisfies the requirement mcts4py-0.16 (from versions: none)
> ERROR: No matching distribution found for mcts4py-0.16
> ```

My apologies @altineel please try this command,
```
pip install -i https://test.pypi.org/simple/ mcts4py
```

**Pull request merged**